biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotationReader.java

```diff
@@ -514,7 +515,7 @@ private void processMethodActivationArgs(String propertyDefKeyFormat, String mem
 				String propertyDefKey = String.format(propertyDefKeyFormat, arg);
 				processActivationObject(propertyDefKey, param, memberDescriptor, details, deactivate);
 			} else if (deactivate && (type == BaseType.I)) {
-				component.updateVersion(V1_1);
+				component.updateVersion(V1_1, "processMethodActivationArgs deactivate && type==BaseType.I");
```

Comment:
An more meaningful reason here is that the deactivate method takes an `int reason` argument.

Reply:
Could you help with the reasons? For me many of the reasons are rather hard to guess from the code.

Reply:
In a way I would be ok with a number that we then describe in more detail somewhere

Reply:
I think that level of indirection will be a source of errors for us to maintain. I can help with the reason strings.